src/viam/sdk/tests/mocks/mlmodel_mocks.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright 2023 Viam Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <viam/sdk/tests/mocks/mlmodel_mocks.hpp>
+
+namespace viam {
+namespace sdktests {
+
+MockMLModelService& MockMLModelService::set_infer_handler(infer_handler handler) {
```

Comment:
(q) It doesn't look to me like the return value of this is used in the tests. Curious why you have it return the ptr instead of being void.

Reply:
Just because I like setters that chain:

`foo.set_x(42).set_y(true)`.

So I tend to write them that way, but in this case I didn't end up using it. I can remove it if you aren't in favor.